Update documentation

ASPetrov opened the pull request.

Commit:
Add branching strategy

Commit:
[FIX] Code-review.md was left empty

Commit:
[FIX] Broken link to code style guideline

senart commented on GIT-BRANCHING-STRATEGY.md:
```diff
@@ -0,0 +1,123 @@
+
+## Merging
+
+All merges into `develop`, `release/*`, and `hotfix/*` should happen via pull requests. This ensures that all code gets reviewed at some point before it's shipped. This ensures that all code gets reviewed at some point before it's shipped. For more information, see [Code Review Standards](CODE-REVIEW.md).
```
Instead of copying `CODE-REVIEW.md` in this repo, wouldn't it be better to just [provide a link](https://github.com/thoughtbot/guides/tree/master/code-review)?

senart commented on GIT-BRANCHING-STRATEGY.md:
```diff
@@ -0,0 +1,123 @@
+
+We consider `origin/develop` to be the main branch where the source code of HEAD always reflects a state with the latest delivered development changes for the next release. Some would call this the “integration branch”. This is also where any automatic nightly builds are built from.
+
+> **NB! When code in the `develop` branch is stable and ready to release, all of the changes should be merged back into master somehow and then tagged with a release number (See [Release branches](#release-branches)).**
```
You can omit "somehow".

Commit:
Remove CODE-REVIEW.md
Replace code review link with remote one in GIT-BRANCHING-STRATEGY.md
[FIX] Some wording in GIT-BRANCHING-STRATEGY.md
